index.d.ts

```diff
@@ -0,0 +1,215 @@
+import {ObisMeasurement} from './lib/ObisMeasurement';
+
+export * as ObisNames from './lib/ObisNames';
+export * as SmlUnits from './lib/protocols/SmlUnits';
+export {ObisMeasurement}
+
+export type ObisOptions = ObisBaseOptions & ObisProtocolOptions & ObisTransportOptions;
+export type ObisLanguage = 'en' | 'de';
+
+
+interface ObisBaseOptions {
+    /**
+     * optional, number of seconds to wait for next request or pause serial receiving, value 0 possible to restart directly after finishing one message, Default: is 300 (=5 Minutes)
+     */
+    requestInterval?: number;
+
+    obisNameLanguage?: ObisLanguage;
+
+    /**
+     * optional, if smartmeter do not return complete OBIS IDs (without medium info) this will be used as fallback for name resolving
+     */
+    obisFallbackMedium?: number;
+
+    /**
+     * optional, values: 0 (no logging), 1 (basic logging), 2 (detailed logging), Default: 0
+     */
+    debug?: 0 | 1 | 2;
+
+    /**
+     * optional, logging function that accepts one parameter to log a string. Default is "console.log"
+     */
+    logger?: (...obj: any[]) => void;
```

Comment:
Just did a little research, `(...obj: any[]) => void` is in fact fully equivalent to `Function` (in TypeScript, `void` is a superset of _any_ return type). Therefore, my intended limitation to `void` returns was useless either way, so that I can happily replace it by `Function`!